src/BloomExe/Publish/EpubMaker.cs

```diff
@@ -137,11 +138,25 @@ public void StageEpub(bool publishWithoutAudio = false)
 					CopyStyleSheets(pageDom);
 			}
 
-			const string coverPageImageFile = "thumbnail-256.png";
+			string coverPageImageFile = "thumbnail-256.png";
 			// This thumbnail is otherwise only made when uploading, so it may be out of date.
 			// Just remake it every time.
 			_thumbNailer.MakeThumbnailOfCover(Book, 256, Form.ActiveForm);
-			CopyFileToEpub(Path.Combine(Book.FolderPath, coverPageImageFile));
+			var coverPageImagePath = Path.Combine(Book.FolderPath, coverPageImageFile);
+			if (!File.Exists(coverPageImagePath))
+			{
+				// Something went wrong...should have been reported already...use the low-res image we always have
```

Comment:
I agree. The bug is a total mystery. I don't see how it's possible to fail to get the thumbnail without getting a crash resulting from the thumbnail generation timing out.
This is just an attempt to mitigate the problem. In 99.9%(?) of the 0.1%(?) of cases where this mysteriously happens, this will make it use the low-res thumb for a cover page, so the user will get an epub...but maybe not one he likes. If TWO completely unexpected failures occur in the same place, we crash (which the timeout code already does, for the small percentage of cases where we can't make a thumbnail of one page in 15 seconds).
Maybe it would be better to display the NonFatalError here (and there), but what do we do then? If we don't crash and also don't produce an epub (which we can't without a cover page graphic), what do we show the user who is expecting a preview of his book? It felt to me like a rare enough case to just say the program can't cope and crash. If not, I expect users will tell us, and maybe we'll get enough more information to fix it properly.
But I can take the extra time to use NonFatalError and replace the Epub preview with some sort of message and disable the save Epub button if you like...